frontend/src/utils/polls/videos.ts

```diff
@@ -11,22 +12,22 @@ let VIDEOS: Promise<Recommendation[]> | null = null;
  *   - are safe recommendations
  *   - are in the TOP 100 of all-time recommendations...
  *   - ...having a maximum duration of 5 minutes
+ *   - are filtered according to the user's navigator languages
  *
  * TODO:
- *   - make the recommendations' language match the user language
- *   - if no reco. are available in this language, use `en`?
+ *   - if no reco. are available in the navigator languages, use `en`?
  */
 export function getTutorialVideos(): Promise<Recommendation[]> {
   if (VIDEOS != null) {
     return VIDEOS;
   }
+  const metadata: Record<string, string | string[]> = {};
 
   const minutesMax = 5;
   const top = 100;
 
-  const metadata = {
-    'duration:lte:int': 60 * minutesMax,
-  };
+  metadata['duration:lte:int'] = (60 * minutesMax).toString();
+  metadata['language'] = recommendationsLanguagesFromNavigator().split(',');
```

Comment:
Wouldn't it be better to do something like in the recommendation page?

```
let loadedLanguages = loadRecommendationsLanguages();

if (loadedLanguages === null) {
  loadedLanguages = recommendationsLanguagesFromNavigator();
  saveRecommendationsLanguages(loadedLanguages);
}
```

For the scenario where a user first plays with recommendations, then goes to the tutorial?

I think we will want to give access to the tutorial even to users that have been on Tournesol for some time already. For example if we make an advance tutorial for new complex feature.

Reply:
Good question.

I first thought about having the same behaviour on both the recommendation page and the tutorial, but finally I decided to propose a more simple approach for the tutorial.

The current implementation avoids the following problem:
- Italian-speaking users go to the recommendation page
- they play with the language filter, to see if Tournesol recommends video in Korean
- they finally start the tutorial
- suggested videos are in Korean, but they speak only Italian

This is not a problem right now, as the tutorial is not easily accessible, the link is only displayed once after the e-mail activation, so back and forth between tutorial and the recommendation page is very very uncommon ; but this will be a problem (maybe a small problem?) when the tutorial link will be displayed on the home page.

> I think we will want to give access to the tutorial even to users that have been on Tournesol for some time already. For example if we make an advance tutorial for new complex feature.

This is a good idea, but do we have already clear ideas about this new tutorial?

The comparison series allow use to create different kind of series, including tutorials, so it's even possible to have a very basic friendly tutorial for new users, using navigators' languages, and a more complete one for "old" users using the recommendation languages.

The only reason that makes me want to keep the implementation as-is, is to potentially avoid confusing new users. But I understand we may want to rework this in the future (also, it could be a good idea to have a language selector directly in the tutorial).

What is your opinion in this @lfaucon ? Would you like to sync the recommendations languages with the tutorial?

Reply:
> This is a good idea, but do we have already clear ideas about this new tutorial?

No

> suggested videos are in Korean, but they speak only Italian

Good point, I did not anticipate this potential issue

> What is your opinion in this @lfaucon ? Would you like to sync the recommendations languages with the tutorial?

I'm convinced about your argument for not syncing both

I realize that even relying on browser language may break, because for many language we don't have recommended videos.

What about hard coding for the moment: "if languages don't contain fr or en, then add en (or selected interface language)" ? We can justify this as the website is only available in fr/en.

Reply:
> What about hard coding for the moment: "if languages don't contain fr or en, then add en (or selected interface language)" ? We can justify this as the website is only available in fr/en.

It sounds reasonable to me.

Reply:
Addressed by 2553f953ca24c61dc1ac8e7e63661929519e20ad.